### ISSUE TITLE
Changing words adds undesired trailing character (incompatible with an autocmd)

### ISSUE THREAD

**Issue by awvalenti** [mg979/vim-visual-multi#215]
**Describe the issue:**

vim-visual-multi does not play well with:
```vim
autocmd InsertLeave * :normal! `^
```

This autocmd makes cursor not go backwards when you leave insert mode. I.e., with it, you can `i<Esc>i<Esc>` and the cursor stays still.

However, when used with vim-visual-multi, it makes it repeat the character after an edition. See GIF below.

![vim](https://user-images.githubusercontent.com/834921/185803852-0fcf73be-ae09-4f30-ab10-dd88e1406355.gif)

**Steps to reproduce**  
1. Run the autocmd above
2. Write some repeated words, finishing them with `!`
3. Activate multi cursor on them, then `c` to change them all together. When you `Esc`, the `!` will get repeated.

-----

- **Operating System**: Windows 10 Pro (probably every OS)
- **Vim Version**: Both of these:
  - VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Jun 23 2022 04:01:15)
Correções incluídas: 1-5117
  - VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Nov 16 2021 23:12:44)
MS-Windows 32-bit GUI version with OLE support

- **commit SHA/branch**: master (e23b98a8852255766e54bf7723a9d61fb5ab3143)

**Comment by mg979** [comment]
VM needs autocommands to work. When exiting insert mode, it will perform an update because some stuff happens just after exiting (abbreviations for example). I cannot avoid using InsertLeave without breaking a lot of stuff.

Try using a timer to restore position.

    autocmd InsertLeave * call timer_start(10, { t -> execute('normal! `^') })

**Comment by awvalenti** [comment]
Cool, that almost worked! But it messes up when I use Backspace.

Isn't there a variable whose value I can read to determine whether we're in
Visual-Multi-Insertion mode? vim-airline calls it V-I. I could then add it
as a condition to run or not the command.

Em dom., 21 de ago. de 2022 14:53, mg979 ***@***.***>
escreveu:

**Comment by mg979** [comment]
```vim
autocmd InsertLeave * if !exists('b:visual_multi') | exe 'normal! `^' | endif
```

**Comment by awvalenti** [comment]
Awesome, thanks, it worked!

(closes #215)

Em ter., 23 de ago. de 2022 17:37, mg979 ***@***.***>
escreveu:
